File: Baekjoon/누적합/1806.py
# https://www.acmicpc.net/problem/1806

import sys
input = sys.stdin.readline

n, s = map(int, input().split())
arr = list(map(int, input().split()))

start = 0
end = 0
total = 0
min_length = 100001

while True:
    if total >= s:
        total -= arr[start]
        start += 1
        min_length = min(min_length, end - start + 1)
    elif end == n:
        break
    else:
        total += arr[end]
        end += 1

if min_length == 100001:
    print(0)
else:
    print(min_length)

File: Baekjoon/누적합/1806.cpp
// https://www.acmicpc.net/problem/1806

#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int n, s;
    cin >> n >> s;
    vector<int> arr(n);
    for (int i = 0; i < n; i++) {
        cin >> arr[i];
    }

    int left = 0, right = 0, sum = arr[0], len = n + 1;

    while (left <= right && right < n) {
        if (sum < s) {
            right++;
            sum += arr[right];
        } else if (sum == s) {
            len = min(len, right - left + 1);
            right++;
            sum += arr[right];
        } else if (sum > s) {
            len = min(len, right - left + 1);
            sum -= arr[left];
            left++;
        }
    }

    if (len == n + 1) {
        cout << 0 << '\n';
    } else {
        cout << len << '\n';
    }

    return 0;
}